src/main/java/com/limechain/teavm/HttpRequest.java

```diff
@@ -1,8 +1,32 @@
 package com.limechain.teavm;
 
+import org.teavm.interop.Async;
+import org.teavm.interop.AsyncCallback;
 import org.teavm.jso.JSBody;
+import org.teavm.jso.JSFunctor;
+import org.teavm.jso.JSObject;
+import org.teavm.jso.core.JSError;
+
+import java.io.IOException;
 
 public class HttpRequest {
-    @JSBody(params = {"method", "url", "body"}, script = "return httpRequestSync(method, url, body);")
-    public static native String httpRequestSync(String method, String url, String body);
+    @JSFunctor
+    interface HttpRequestCallback extends JSObject {
```

Comment:
```suggestion
    private interface HttpRequestCallback extends JSObject {
```
Imo this can be private as it won't be used outside the http request scope.
Also just a preference, I tend to keep inner classes/interfaces either at the end of a file or under the code blocks that actually use it.